### DISCUSSION TITLE
Consuming an encrypted stream from Axis camera

### DISCUSSION THREAD

**Discussion by tykling** [bluenviron/mediamtx#1499] [Questions]
Hello, and thank you for a fantastic product.

I am trying to get encrypted video out of my Axis cameras. I can play the video with ffplay:

```
<snip>
Input #0, matroska,webm, from 'https://root:hunter12@cam.example.com/axis-cgi/media.cgi?streamprofile=myprofile':
  Metadata:
    encoder         : GStreamer matroskamux version 1.16.1
    creation_time   : 2023-02-21T05:43:46.199808Z
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0(eng): Video: h264 (Baseline), yuvj420p(pc, bt709, progressive), 720x1280, SAR 1:1 DAR 9:16, 25 fps, 25 tbr, 1k tbn, 2k tbc (default)
    Metadata:
      title           : Video
<snip>
```

But I have been unable to use this as a source, since it is interpreted as a HLS source because it is https I guess. Question 1) Can I somehow use the HTTPS webm stream as a source directly?

I can also get the video out of the cameras as rtsps on port 322 but I am having trouble playing it - both with ffplay and as a source. `ffplay` in verbose shows this SDP, but I have no idea how to interpret it. Question 2: Is it normal that it says `rtsp://` and not `rtsps://` in the `a:control` lines in there? If not then maybe the issue is inside the camera? Some of the ffplay output when trying to play rtsps directly from the camera is included here:

```
v=0
o=- 273740523035808301 1 IN IP4 192.168.40.81
s=Session streamed with GStreamer
i=rtsp-server
t=0 0
a=tool:GStreamer
a=type:broadcast
a=range:npt=now-
a=control:rtsp://cam.example.com:322/axis-media/media.amp
m=video 0 RTP/SAVP 96
c=IN IP4 0.0.0.0
b=AS:50000
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1;profile-level-id=420029;sprop-parameter-sets=Z0IAKeKQFoFDTcBAQGkHiRFQ,aM48gA==
a=key-mgmt:mikey AQAFAHRzqc0BAAC9kLbRAAAAAAsA5<snip>YhLn/sR3M/xPoV9O0jcvcS1G4HVAA==
a=ts-refclk:local
a=mediaclk:sender
a=recvonly
a=control:rtsp://cam.example.com:322/axis-media/media.amp/stream=0
a=framerate:25.000000
a=transform:0.000000,1.000000,0.000000;-1.000000,0.000000,0.000000;0.000000,0.000000,1.000000

rtsps://hunter12@cam.example.com/axis-media/media.amp: Server returned 400 Bad Request
```

I guess I could runOnInit ffmpeg and use that to pull the stream, instead of using it as a source directly. But if it was possible to pull the stream directly without ffmpeg I would prefer that to minimize latency.

The cameras are Axis `P3935-LR` and `Axis M3085-V` in case that is interesting to anyone.

I have been at this for a couple of days, I would love someone with more knowledge about streaming than me to clue me up. Thanks a million! :) <3

**Comment by tykling** [comment]
Axis support gave me this https://www.axis.com/ftp/devcom_files/manuals/Axis_SRTP_Integration_Guide.pdf which I am not sure what to do with

**Comment by aler9** [comment]
Hello, my advice is to use the RTSPS protocol, since the raw HTTP protocol is not supported and even if it it was supported, it would have a greater latency than RTSP.

Try using the `rtsps://` url as source directly inside the server, without FFmpeg. If the server is unable to read it, open an issue and follow instructions in order to provide relevant data that allows to replicate the bug (server logs and/or network dump).

> Is it normal that it says rtsp:// and not rtsps:// in the a:control lines in there? 

It should say `rtsps://` but the library behind MediaMTX / rtsp-simple-server ignores it and uses RTSPS anyway. I don't know what FFmpeg does but i don't care since this server has the aim of ingesting RTSPS directly without FFmpeg.

[a comment by an automated account is not shown]
